File: Cpp/495_Teemo_Attacking.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int result = 0;
        if(timeSeries.empty()) return result;
        
        for(int i = 1; i<timeSeries.size(); ++i) {
            result += min(timeSeries[i] - timeSeries[i-1], duration);
        }
        result += duration;
        
        return result;
    }
};

File: Kotlin/495_Teemo_Attacking.kt
class Solution {
    fun findPoisonedDuration(timeSeries: IntArray, duration: Int): Int {
        var result = 0
        for(i in 1 until timeSeries.size) {
            result += minOf(timeSeries[i] - timeSeries[i-1], duration)
        }
        result += duration
        
        return result
    }
}